CRM/Mailing/Form/Unsubscribe.php

```diff
@@ -78,9 +78,10 @@ public function preProcess() {
     $this->_email = $email;
 
     $groups = CRM_Mailing_Event_BAO_MailingEventUnsubscribe::unsub_from_mailing($job_id, $queue_id, $hash, TRUE);
-    $this->assign('groups', $groups);
+    $this->assign('groups', $groups ?? []);
```

Comment:
This cast to array is more about consistency - I don't think it tries to process it if empty due the nasty groupExist pattern